src/main/com/algos/src/datastructures/caches/LRUCache.java

```diff
@@ -120,52 +107,12 @@ private void addNewEntry(Entry<K, V> newEntry) {
     }
 
     static final class Entry<I, J> {
-
         private Entry<I, J> preEntry;
         private Entry<I, J> nextEntry;
         private I key;
         private J value;
 
         public Entry() {
-        }
-
-        public Entry(Entry<I, J> preEntry, Entry<I, J> nextEntry, I key, J value) {
-            this.preEntry = preEntry;
-            this.nextEntry = nextEntry;
-            this.key = key;
-            this.value = value;
-        }
-
-        public Entry<I, J> getPreEntry() {
-            return preEntry;
-        }
-
-        public void setPreEntry(Entry<I, J> preEntry) {
-            this.preEntry = preEntry;
-        }
-
-        public Entry<I, J> getNextEntry() {
-            return nextEntry;
-        }
-
-        public void setNextEntry(Entry<I, J> nextEntry) {
-            this.nextEntry = nextEntry;
-        }
-
-        public I getKey() {
-            return key;
-        }
-
-        public void setKey(I key) {
-            this.key = key;
-        }
-
-        public J getValue() {
-            return value;
-        }
-
-        public void setValue(J value) {
-            this.value = value;
+            
         }
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/rishabhbohra3/Algo-tutorial-test/pull/6#issuecomment-2155542587>#2011d9</a><span> - 07/06/2024, 09:06 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The removal of constructors and getter/setter methods for the Entry class significantly reduces the encapsulation and flexibility of the Entry objects. This change could lead to direct manipulation of Entry object fields from outside the class, which is a poor practice in object-oriented design.
</div>
<div id="fix">
<b>Fix:</b> Reintroduce the constructors, getters, and setters for the Entry class to ensure proper encapsulation and maintainability.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div>
<pre><code>
 @@ -110,7 +110,21 @@
 +        public Entry() {
 +        }
 +
 +        public Entry(Entry&lt;I, J&gt; preEntry, Entry&lt;I, J&gt; nextEntry, I key, J value) {
 +            this.preEntry = preEntry;
 +            this.nextEntry = nextEntry;
 +            this.key = key;
 +            this.value = value;
 +        }
 +
 +        public Entry&lt;I, J&gt; getPreEntry() {
 +            return preEntry;
 +        }
 +
 +        public void setPreEntry(Entry&lt;I, J&gt; preEntry) {
 +            this.preEntry = preEntry;
 +        }
 +
 +        public Entry&lt;I, J&gt; getNextEntry() {
 +            return nextEntry;
 +        }
 +
 +        public I getKey() {
 +            return key;
 +        }
 +
 +        public void setKey(I key) {
 +            this.key = key;
 +        }
 +
 +        public J getValue() {
 +            return value;
 +        }
 +
 +        public void setValue(J value) {
 +            this.value = value;
 +        }
</code></pre>
</div>
</details>
</div>
<br>


</div>